### PR TITLE
fix: Fix handling of expressions in `values()`/`values_list()`

### LINKED ISSUE #670: Error "'Cast' object has no attribute 'split'" when using Cast in values_list()
We're using wagtail with [wagtail-modeltranslation](https://github.com/infoportugal/wagtail-modeltranslation), which uses [django-modeltranslation](https://github.com/deschler/django-modeltranslation), and we've stumbled over this error:

    2023-02-08 16:41:30 ERROR django.request: Internal Server Error: /admin/ (log.py:224)
    Traceback (most recent call last):
      File "***/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
        response = get_response(request)
      File "***/lib/python3.8/site-packages/django/core/handlers/base.py", line 204, in _get_response
        response = response.render()
      File "***/lib/python3.8/site-packages/wagtail/admin/auth.py", line 197, in overridden_render
        return render()
      ...
      File "***/lib/python3.8/site-packages/django/template/library.py", line 192, in render
        output = self.func(*resolved_args, **resolved_kwargs)
      File "***/lib/python3.8/site-packages/wagtail/admin/templatetags/wagtailadmin_tags.py", line 920, in component
        return obj.render_html(context)
      File "***/lib/python3.8/site-packages/wagtail/admin/ui/components.py", line 15, in render_html
        context_data = self.get_context_data(parent_context)
      File "***/lib/python3.8/site-packages/wagtail/admin/views/home.py", line 103, in get_context_data
        user_perms.revisions_for_moderation()
      File "***/lib/python3.8/site-packages/wagtail/models/__init__.py", line 2953, in revisions_for_moderation
        object_id__in=Page.objects.filter(only_my_sections).values_list(
      File "***/lib/python3.8/site-packages/modeltranslation/manager.py", line 450, in values_list
        clone = self._values(*fields, prepare=True, selects_all=selects_all)
      File "***/lib/python3.8/site-packages/modeltranslation/manager.py", line 419, in _values
        clone = super(MultilingualQuerySet, self)._values(*list(new_fields), **kwargs)
      File "***/lib/python3.8/site-packages/django/db/models/query.py", line 835, in _values
        clone.query.set_values(fields)
      File "***/lib/python3.8/site-packages/django/db/models/sql/query.py", line 2272, in set_values
        self.add_fields(field_names, True)
      File "***/lib/python3.8/site-packages/django/db/models/sql/query.py", line 1955, in add_fields
        join_info = self.setup_joins(name.split(LOOKUP_SEP), opts, alias, allow_many=allow_m2m)
    AttributeError: 'Cast' object has no attribute 'split'

django-modeltranslation does not yet support `values()` or `values_list()` calls with expressions, like django's `Cast(...)`-operator ([docs](https://docs.djangoproject.com/en/3.2/ref/models/database-functions/#cast)) which is used by wagtail above.

I've already fixed this locally and I will make a PR shortly.

### PR DESCRIPTION
**Pull request by th3hamm0r**
This basically duplicates some code from django's implementation.

Fixes #670

### CODE REVIEW

File: modeltranslation/manager.py [hunk 1]
```diff
@@ -447,7 +448,25 @@ def values_list(self, *fields, flat=False, named=False):
         if not fields:
             # Emulate original queryset behaviour: get all fields that are not translation fields
             fields = self._get_original_fields()
-        clone = self._values(*fields, prepare=True, selects_all=selects_all)
+
```
Comment (th3hamm0r) [inline modeltranslation/manager.py:451]: Fyi: this is based on https://github.com/django/django/blob/2fd755b361d3da2cd0440fc9839feb2bb69b027b/django/db/models/query.py#L1318-L1337

File: modeltranslation/manager.py [hunk 2]
```diff
@@ -429,6 +429,7 @@ def values(self, *fields, **expressions):
         if not fields:
             # Emulate original queryset behaviour: get all fields that are not translation fields
             fields = self._get_original_fields()
+        fields += tuple(expressions)
```
Comment (th3hamm0r) [inline modeltranslation/manager.py:432]: This is based on https://github.com/django/django/blob/2fd755b361d3da2cd0440fc9839feb2bb69b027b/django/db/models/query.py#L1304

File: modeltranslation/manager.py [hunk 3]
```diff
@@ -447,7 +448,25 @@ def values_list(self, *fields, flat=False, named=False):
...
+                    field_id = field_id_prefix + str(counter)
+                    counter += 1
+                    if field_id not in field_names:
+                        break
+                expressions[field_id] = field
+                _fields.append(field_id)
+            else:
+                _fields.append(field)
+
+        clone = self._values(*_fields, prepare=True, selects_all=selects_all, **expressions)
```
Comment (th3hamm0r) [inline modeltranslation/manager.py:469]: Fyi: this is based on https://github.com/django/django/blob/2fd755b361d3da2cd0440fc9839feb2bb69b027b/django/db/models/query.py#L1318-L1337

**Comment by last-partizan** [comment]
LGTM, merging.

Thanks!

**Pull request merged**